### PR TITLE
Add Kit4DL

### PR DESCRIPTION
**Pull request by jamesWalczak**
Checklist
- [x] Title of this PR is meaningful: e.g. "Adding my_nifty_package", not "updated meta.yaml".
- [x] License file is packaged (see [here](https://github.com/conda-forge/staged-recipes/blob/5eddbd7fc9d1502169089da06c3688d9759be978/recipes/example/meta.yaml#L64-L73) for an example).
- [x] Source is from official source.
- [x] Package does not vendor other packages. (If a package uses the source of another package, they should be separate packages or the licenses of all packages need to be packaged).
- [x] If static libraries are linked in, the license of the static library is packaged.
- [x] Package does not ship static libraries. If static libraries are needed, [follow CFEP-18](https://github.com/conda-forge/cfep/blob/main/cfep-18.md).
- [x] Build number is 0.
- [x] A tarball (`url`) rather than a repo (e.g. `git_url`) is used in your recipe (see [here](https://conda-forge.org/docs/maintainer/adding_pkgs.html#build-from-tarballs-not-repos) for more details).
- [x] GitHub users listed in the maintainer section have posted a comment confirming they are willing to be listed there.
- [x] When in trouble, please check our [knowledge base documentation](https://conda-forge.org/docs/maintainer/knowledge_base.html) before pinging a team.

### CODE REVIEW

**Comment by jamesWalczak** [comment]
@conda-forge/help-python I need help with pipelines.  Job `Run docker build` raises `subprocess.CalledProcessError: Command 'cp -a /home/conda/staged-recipes-copy/recipes/.... returned non-zero exit status 1.` but I cannot find the reason.

File: recipes/kit4dl/meta.yaml
Comment (synapticarbors) [inline recipes/kit4dl/meta.yaml:18]: I do not believe you need the `build` section  under `requirements`

File: recipes/kit4dl/meta.yaml
Comment (synapticarbors) [inline recipes/kit4dl/meta.yaml:24]: remove quotes from around requirements

File: recipes/kit4dl/meta.yaml
Comment (synapticarbors) [inline recipes/kit4dl/meta.yaml:33]: conda doesn't support package "extras" like `[all]`. You would need to add the requirements here. See:
https://github.com/tiangolo/typer/blob/master/pyproject.toml#L67-L71

Note: These are the requirements from the latest revision on Github, so you should check to see if the version you're attempting to build from is the same.

File: recipes/kit4dl/meta.yaml
Comment (synapticarbors) [inline recipes/kit4dl/meta.yaml:35]: I would recommend adding a `pip check` test:
https://github.com/conda-forge/staged-recipes/blob/main/recipes/example/meta.yaml#L61-L64

File: recipes/kit4dl/meta.yaml
Comment (synapticarbors) [inline recipes/kit4dl/meta.yaml:12]: It looks like this package has a command line entry point, so you'll want to add the following under `build`.

```
entry_points:
    - kit4dl = kit4dl.cli.app:run
 ```

And then under `tests` > `commands`, test the cli:

```
commands:
    - pip check
    - kit4dl --help
requires:
    - pip
 ```

**Review by synapticarbors — CHANGES_REQUESTED** [review]

**Comment by synapticarbors** [comment]
It's unclear to me why this is generating the error on `cp` that's in the logs. @conda-forge/staged-recipes -- any idea what might be causing this?

**Comment by ocefpaf** [comment]
This copy location is odd:

```
/home/conda/staged-recipes-copy/recipes/kit4dl/https:/pypi.io/packages/source/k/kit4dl/kit4dl-2023.08b0.tar.gz
```

File: recipes/kit4dl/meta.yaml
Comment (ocefpaf) [inline recipes/kit4dl/meta.yaml:2]: This one falls under https://github.com/conda-forge/cfep/blob/main/cfep-05.md

File: recipes/kit4dl/meta.yaml
Comment (ocefpaf) [inline recipes/kit4dl/meta.yaml:28]: The metadata says:

```suggestion
    - numpy <=1.25.0
```

not sure what is correct here though. I'd raise a question upstream.

File: recipes/kit4dl/meta.yaml
Comment (ocefpaf) [inline recipes/kit4dl/meta.yaml:29]: Same situation as numpy, the metadata says:

```suggestion
    pydantic <2.0.0
```

File: recipes/kit4dl/meta.yaml
Comment (ocefpaf) [inline recipes/kit4dl/meta.yaml:35]: These are not listed in the metadata.

```suggestion
```

File: recipes/kit4dl/meta.yaml
Comment (ocefpaf) [inline recipes/kit4dl/meta.yaml:36]: Metadata has toml there.

```suggestion
    - typer
    - toml
```

File: recipes/kit4dl/meta.yaml
Comment (synapticarbors) [reply recipes/kit4dl/meta.yaml:35]: I suggested adding these because in the metadata it's `typer[all]`, which implies:

```
https://github.com/tiangolo/typer/blob/master/pyproject.toml#L67-L70
```

Comment (synapticarbors) [reply recipes/kit4dl/meta.yaml:35]:
```
all = [
    "colorama >=0.4.3,<0.5.0",
    "shellingham >=1.3.0,<2.0.0",
    "rich >=10.11.0,<14.0.0",
]
```

File: recipes/kit4dl/meta.yaml
Comment (jamesWalczak) [reply recipes/kit4dl/meta.yaml:36]: yes, but it's needed for python 3.10. for python 3.11 there is builtint library `tomlib`. anyway , i can add this.

**Comment by jamesWalczak** [comment]
I accepted the changes you proposed, but still, the issue about copy fail persists. Any ideas, what can be the reason?

**Comment by ocefpaf** [comment]
> I accepted the changes you proposed, but still, the issue about copy fail persists. Any ideas, what can be the reason?

Those were not related to that error. I don't know why that is happening BTW. It is very odd :-/

**Comment by jamesWalczak** [comment]
@conda-forge/help-python I fixed the issue, however, there is still a problem with PyTorch dependencies for Windows, could you help with that one?

**Comment by ocefpaf** [comment]
> @conda-forge/help-python I fixed the issue, however, there is still a problem with PyTorch dependencies for Windows, could you help with that one?

The problem is that pytroch is not available on Windows. There is nothing you can do about that, unless you know how to build it on Windows ;-p

**Comment by jamesWalczak** [comment]
@ocefpaf it seems it can be built on Windows according to the README file of pytorch repo. see  *from source* section   https://github.com/pytorch/pytorch#from-source

**Comment by ocefpaf** [comment]
> @ocefpaf it seems it can be built on Windows according to the README file of pytorch repo. see _from source_ section https://github.com/pytorch/pytorch#from-source

Oh. We know it can be done. The question is who can do it. Anyways, that is not a blocker here and your package should remain noarch. @synapticarbors was your first reviewer here, so let's wait for his final decision and merge.

**Comment by jamesWalczak** [comment]
@synapticarbors the pull request is ready to review.

**Pull request merged**
